scripts/ingests/calamari/ingest_calamari_spt

```diff
@@ -0,0 +1,284 @@
+    guess=False,
+    fast_reader=False, 
+    delimiter=",",
+)
+
+spectral_types_ingested = 0
+sources_ingested = 0
+spectral_types_already_exist = 0
+binaries = [
+    "HD 130948BC",
+    "HD 130948C",
+    "HD 130948B",
+    "Gl 337CD",
+    "Gl 337C",
+    "Gl 337D",
+    "Gl 417BC",
+    "Gl 417B",
+    "Gl 417C"
+]
+
+#helper method to retrieve the publication links from calamari_data
+def getRef(ref_index):
+    ref = ref_index.split(',')[0]
+    ref_link = ref_table[int(ref)]['ADS']
+    if 'iopscience' not in ref_link or 'harvard.edu' not in ref_link:
+        ref_link = ref_table[int(ref)]['Link']
+    return ref_link
+
+#helper method to retrieve the bibcode from a link
+def extractADS(link):
+    start = link.find('abs/')+4
+    end = link.find('/abstract')
+    ads = link[start:end]
+    ads = ads.replace("%26", "&")
+    return ads
+
+#helper method to retrieve the doi from a link
+def extractDOI(link):
+    link = str(link)
+    if 'iopscience' in link:
+        start = link.find('article/')+8
+        doi = link[start:]
+        doi = doi.replace("/pdf", "")
+    else:
+        start = link.find('doi.org/')+8
+        doi=link[start:]
+    return doi
+
+def otherReferencesList(ref):
+    #get all the ids/indexes of the references
+    ids = ref.split(", ")
+    result = []
+    #for each reference...
+    for id in ids:
+        link = ref_table[int(id)]['ADS']
+        #if bibcode or doi is not directly in the link... go to Link column
+        if 'iopscience' not in link or 'harvard.edu' not in link:
+            link = ref_table[int(id)]['Link']
+        #if bibcode is directly in the link
+        if 'harvard.edu' in link:
+            bibcode = extractADS(link)
+            pub_result=find_publication(
+                db=db,
+                bibcode=bibcode
+            )
+            if pub_result[0]:
+                result.append(pub_result[1])
+            else:
+                print(f"Warning: Publication not found for bibcode {bibcode}")
+            #if doi code is found directly in the link
+        elif 'iopscience' in link or 'doi.org' in link:
+            doi=extractDOI(link)
+            pub_result=find_publication(
+                db=db,
+                doi=doi
+            )
+            if pub_result[0]:
+                result.append(pub_result[1])
+            else:
+                print(f"Warning: Publication not found for doi {doi}")
+        #use reference name to find reference
+        else:
+            reference= ref_table[int(id)]['Ref']
+            reference= reference.replace("+", "")
+            reference=reference[0:4] + reference[-2:]
+            pub_result=find_publication(
+                db=db,
+                reference=reference
+            )
+            if pub_result[0]:
+                result.append(pub_result[1])
+            else:
+                print(f"Warning: Publication not found for reference {reference}")
+    #return list of references
+    return result
```

Comment:
make a dedicated calamari_refs module and import rather than duplicating code in two scripts.